src/validation.cpp

```diff
@@ -465,6 +463,114 @@ static bool CheckInputsFromMempoolAndCache(const CTransaction& tx, TxValidationS
     return CheckInputScripts(tx, state, view, flags, /* cacheSigStore = */ true, /* cacheFullSciptStore = */ true, txdata);
 }
 
+/** An empty coin used as a placeholder for a spent coin.*/
+static const Coin coin_spent;
+/**
+ * A CoinsView that adds a memory cache to another CoinsView and serves as temporary scratch space.
+ * Used by MemPoolAccept class to validate transactions and packages before submitting to mempool.
+ * A backend can be set to provide read access to chainstate and/or mempool coins, but writing to
+ * the backend is disabled. Avoid using a CCoinsViewTemporary in consensus-critical paths such
+ * as writing to the script cache. See CheckInputsFromMempoolAndCache as an example. When not being
+ * used to validate a package (m_temp_added and m_temp_spent are empty), a CCoinsViewTemporary
+ * behaves exactly like a CCoinsViewCache.
+ */
+class CCoinsViewTemporary : public CCoinsViewCache
```

Comment:
Code-organisation wise, maybe better to move it in `coins.h`, validation.cpp is already wide enough ?

Reply:
I actually think it's better organization to put it in validation.cpp, since it's only used by MemPoolAccept.